### PR TITLE
Upgrade handler

### PR DESCRIPTION
**Pull request by vuong177**
## Description
- Upgrade handler for SDK 46.

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by faddat** [comment]
Hey guys, pls don't forget to set the store key for the ephocs module to nil -- iirc that's the right way to remove a module.  I think.  You may wanna double check on that. 

Thanks!

**Comment by faddat** [comment]
Also check out the release nodes please:

https://github.com/cosmos/cosmos-sdk/releases/tag/v0.46.0

**Review by faddat — APPROVED** [review]
STYLE POINTS, FOR UPGRADE.SH

good work Hieu

File: app/upgrade/fork.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		}
+
+		// override here
+		return newVM, err
+	}
+}
+
+// Fixes an error where validators can be created with a commission rate
+// less than the network minimum rate.
+func FixMinCommisionRate(ctx sdk.Context, staking *stakingkeeper.Keeper) {
```
Comment (mustafapylons) [inline app/upgrade/fork.go:34]: "CommissionRate" would be a better spelling for the function.

File: app/upgrade/fork.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+// CreateUpgradeHandler make upgrade handler
+func CreateUpgradeHandler(mm *module.Manager, configurator module.Configurator, staking *stakingkeeper.Keeper, pylonStoreKey storetypes.StoreKey, cdc codec.Codec) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		newVM, err := mm.RunMigrations(ctx, configurator, vm)
+		if err != nil {
+			return newVM, err
+		}
+		FixMinCommisionRate(ctx, staking)
```
Comment (mustafapylons) [inline app/upgrade/fork.go:21]: "CommissionRate" would be a better spelling for the function.

**Review by mustafapylons — COMMENTED** [review]
few typos, otherwise code is good.

File: app/upgrade/fork.go (hunk 2 above)
Comment (vuong177) [reply app/upgrade/fork.go:21]: thanks @mustafapylons

**Review by mustafapylons — APPROVED** [review]
Looks good to me now.

**Pull request merged**